src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -54,58 +58,90 @@ public boolean isRelative() {
 
     private final TapPositionChange tapPositionChange;
 
-    private LfAction(String id, LfBranch disabledBranch, LfBranch enabledBranch, TapPositionChange tapPositionChange) {
+    private final Map<LfBus, Pair<Double, PowerShift>> busesLoadShift;
+
+    private LfAction(String id, LfBranch disabledBranch, LfBranch enabledBranch, TapPositionChange tapPositionChange, Map<LfBus, Pair<Double, PowerShift>> busesLoadShift) {
         this.id = Objects.requireNonNull(id);
         this.disabledBranch = disabledBranch;
         this.enabledBranch = enabledBranch;
         this.tapPositionChange = tapPositionChange;
+        this.busesLoadShift = busesLoadShift;
     }
 
-    public static Optional<LfAction> create(Action action, LfNetwork network) {
+    public static Optional<LfAction> create(Action action, LfNetwork lfNetwork, Network network, boolean breakers) {
         Objects.requireNonNull(action);
         Objects.requireNonNull(network);
         switch (action.getType()) {
             case SwitchAction.NAME:
-                return create((SwitchAction) action, network);
+                return create((SwitchAction) action, lfNetwork);
 
             case LineConnectionAction.NAME:
-                return create((LineConnectionAction) action, network);
+                return create((LineConnectionAction) action, lfNetwork);
 
             case PhaseTapChangerTapPositionAction.NAME:
-                return create((PhaseTapChangerTapPositionAction) action, network);
+                return create((PhaseTapChangerTapPositionAction) action, lfNetwork);
+
+            case LoadAction.NAME:
+                return create((LoadAction) action, lfNetwork, network, breakers);
 
             default:
                 throw new UnsupportedOperationException("Unsupported action type: " + action.getType());
         }
     }
 
-    private static Optional<LfAction> create(PhaseTapChangerTapPositionAction action, LfNetwork network) {
-        LfBranch branch = network.getBranchById(action.getTransformerId()); // only two windings transformer for the moment.
+    private static Optional<LfAction> create(LoadAction action, LfNetwork lfNetwork, Network network, boolean breakers) {
+        Load load = network.getLoad(action.getLoadId());
+        Terminal terminal = load.getTerminal();
+        Bus bus = breakers ? terminal.getBusBreakerView().getBus() : terminal.getBusView().getBus();
+        if (bus != null) {
```

Comment:
For me: I think I have to check that the load is not in contingency. Add a unit test!

Reply:
Maybe test it in the apply method?

Reply:
Yes, I just get what you mean. Yes, just test that in the apply method.